jiant/modules/seq2seq_decoder.py

```diff
@@ -191,16 +273,16 @@ def forward(
 
         output_dict = {"logits": logits}
 
-        if target_tokens:
-            targets = target_tokens["words"]
-            target_mask = get_text_field_mask(target_tokens)
-            output_dict["target_mask"] = target_mask
-            relevant_logits = logits[:, : targets.shape[1] - 1, :].contiguous()
-            loss = self._get_loss(relevant_logits, targets, target_mask)
-            output_dict["loss"] = loss
-        else:
-            # This is needed in case no gold target sequence is available.
-            output_dict["loss"] = torch.tensor([-1.0])
+        targets = target_tokens["words"]
+        target_mask = get_text_field_mask(target_tokens)
+        output_dict["target_mask"] = target_mask
+        relevant_logits = logits[:, : targets.shape[1] - 1, :].contiguous()
+        loss = self._get_loss(relevant_logits, targets, target_mask)
+        output_dict["loss"] = loss
```

Comment:
1) Is there a way to return loss of gold tokens at test time ?

2) Do we never query `output_dict` for the loss at test time ? We're no longer hard-setting it to zero here.

Reply:
Loss is initialized somewhere else in the code to 0.00; this is returned in case we return the beam_search_output earlier.